### PR TITLE
[ENG-3864] provide queue position and estimated wait time

### PR DESCRIPTION
**Pull request by dsokal**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

To provide users with their queue position progress + estimated queue wait time.
This should mostly look the same as on the website. 

# How

This is implemented only when building a build for a single platform. That is, we don't provide the info with `-p all`. This may change in a follow-up PR.

- Display a progress bar when the build is waiting in queue. Update it every few seconds.
- Display estimated wait time (rounded to 10 minutes).
- I refactored some of the functions in build.ts.
- I changed the build command timeout to 130 minutes (120 minutes is the max build time + some margin).
- If build is waiting to enter the queue (the user has exceeded their concurrency limit) print a URL where they can check their limits (account settings).

# Test Plan
Screenshot 2022-04-05 at 12 00 50

Screenshot 2022-04-05 at 13 28 22

Screenshot 2022-04-05 at 14 23 13

### CODE REVIEW

**Comment by dsokal** [comment]
/changelog-entry new-feature Provide queue position progress and estimated wait time.

File: packages/eas-cli/package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "@urql/core": "2.4.3",
     "@urql/exchange-retry": "0.3.2",
     "chalk": "4.1.2",
+    "cli-progress": "3.10.0",
```
Comment (kbrandwijk) [inline packages/eas-cli/package.json:33]: Any particular reason for picking a different progress bar lib than expo-cli? https://www.npmjs.com/package/progress

Comment (kbrandwijk) [inline packages/eas-cli/package.json:33]: Any reason for going with a different lib than expo-cli here?

**Review by kbrandwijk — APPROVED** [review]

File: packages/eas-cli/package.json (hunk 1 above)
Comment (dsokal) [reply packages/eas-cli/package.json:33]: Yes, `progress` is no longer maintained. 
- https://www.npmjs.com/package/progress Last publish: 3 years ago
- https://www.npmjs.com/package/cli-progress Last publish: 3 months ago

File: packages/eas-cli/src/build/build.ts [hunk 2]
```diff
@@ -208,105 +210,205 @@ async function sendBuildRequestAsync<TPlatform extends Platform, Credentials, TJ
...
+    // 2 hours (max build time limit) + 10 minutes (possible queue time )
+    timeoutSec = 2 * 60 * 60 + 10 * 60,
+    intervalSec = 10,
+  } = {}
+): Promise<MaybeBuildFragment[]> {
+  const b = `build${buildIds.length > 1 ? 's' : ''}`;
+  Log.log(`Waiting for ${b} to complete. You can press Ctrl+C to exit.`);
+  const spinner = ora(`Waiting for ${b} to complete.`).start();
+  const endTime = new Date().getTime() + timeoutSec * 1000;
+  while (new Date().getTime() <= endTime) {
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/build.ts:227]:
```suggestion
  const endTime = Date.now() + timeoutSec * 1000;
  while (Date.now() <= endTime) {
```

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**